File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/ide/acl/entities/GithubConnectionExt.java
package com.duxtinto.intellij.plugin.github.codereviews.ide.acl.entities;

import com.google.common.collect.ForwardingObject;
import org.jetbrains.plugins.github.api.GithubConnection;

public class GithubConnectionExt extends ForwardingObject {
    private final GithubConnection delegate;

    private GithubConnectionExt(GithubConnection connection) {
        this.delegate = connection;
    }

    public static GithubConnectionExt create(GithubConnection delegate) {
        return new GithubConnectionExt(delegate);
    }

    @Override
    public GithubConnection delegate() {
        return delegate;
    }
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/ide/acl/entities/GithubConnectionExt.kt
package com.duxtinto.intellij.plugin.github.codereviews.ide.acl.entities

import com.google.common.collect.ForwardingObject
import org.jetbrains.plugins.github.api.GithubConnection

class GithubConnectionExt private constructor(private val delegate: GithubConnection) : ForwardingObject() {

    public override fun delegate(): GithubConnection {
        return delegate
    }

    companion object {
        fun create(delegate: GithubConnection): GithubConnectionExt {
            return GithubConnectionExt(delegate)
        }
    }
}
